SebastianHerrera/HackerrankSpring/src/main/java/com/globant/hackerrankspring/exception/CustomExceptionHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.globant.hackerrankspring.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Sebastian Herrera (bermeosebas@gmail.com)
+ * @created 06/09/21
+ */
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+@ControllerAdvice
+public class CustomExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(Exception.class)
+    public final ResponseEntity<Object> handleAllExceptions(Exception ex, WebRequest request) {
+        List<String> details = new ArrayList<>();
```

Comment:
If you only add one thing into a list, I suggest to you use Collections.singletonList

The reason :
https://www.it-swarm-fr.com/fr/java/arrays.aslist-vs-collections.singletonlist/1049748648/